sdk/rust-language-support/libveracruz/src/host.rs

```diff
@@ -503,3 +513,105 @@ pub fn getrandom(buffer: &mut [u8]) -> HCallReturnCode<()> {
 
     lift(retcode, ()).unwrap()
 }
+
+/// Returns the size of previous result
+pub fn previous_result_size() -> HCallReturnCode<u32> {
+    let retcode;
+    let mut buffer = vec![0u8; 4];
+
+    unsafe {
+        retcode = __veracruz_hcall_previous_result_size(buffer.as_mut_ptr() as *mut u8);
+    };
+
+    lift(retcode, LittleEndian::read_u32(&buffer)).unwrap()
+}
+
+/// Reads the previous result, returned Option<Vec<u8>>.
+/// The program is assumed to understand how to parse the array of bytes, wrapped in Some(-)
+/// returned into something more meaningful, e.g. by fixing an assumed encoding
+/// between data source providers and the program owner.
+pub fn read_previous_result() -> HCallReturnCode<Option<Vec<u8>>> {
+    previous_result_size().and_then(|size| {
+        let mut buffer = vec![0u8; size as usize];
+        let retcode;
+
+        unsafe {
+            retcode = __veracruz_hcall_read_previous_result(buffer.as_mut_ptr() as *mut u8, size)
+        };
+
+        let result: Option<Vec<u8>> = pinecone::from_bytes(&buffer).unwrap();
```

Comment:
This `pinecone` usage should be removed.  This should be implemented analogously to how `read_input()` is implemented.